streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -264,6 +264,72 @@ public void shouldSupportNonPrefixedProducerConfigs() {
         assertEquals(1, configs.get(ProducerConfig.METRICS_NUM_SAMPLES_CONFIG));
     }
 
+    @Test
+    public void shouldSupportCustomNonPrefixedPropertiesThatAreNotPartOfConsumerConfig() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getConsumerConfigs("groupId", "clientId");
+        assertEquals("host", consumerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldSupportCustomNonPrefixedPropertiesThatAreNotPartOfRestoreConsumerConfig() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getRestoreConsumerConfigs("clientId");
+        assertEquals("host", consumerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldSupportCustomNonPrefixedPropertiesThatAreNotPartOfProducerConfig() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> producerConfigs = streamsConfig.getProducerConfigs("clientId");
+        assertEquals("host", producerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldAddCustomNonPrefixedPropertiesToProducerAndConsumerConfig() {
```

Comment:
This seems to be redundant from above (or already the merged test...) -- restore-consumer and adminclient would be missing though.

Reply:
Yes, I'll clean it up.